### ISSUE TITLE
Error creating new registration :: No MX record for domain yaxom.com

### ISSUE THREAD

**Issue by gjb** [letsencrypt/letsencrypt#1499]
I followed the instructions in the email "Let's Encrypt Closed Beta Invite" exactly as given:

    git clone https://github.com/letsencrypt/letsencrypt
    cd letsencrypt
    ./letsencrypt-auto --agree-dev-preview --server https://acme-v01.api.letsencrypt.org/directory certonly

Near the end of the process I was given a popup saying "Enter email address (used for urgent notices and lost key recovery)". I entered my email address as used for the beta invite and was then told:

    An unexpected error occurred.
    Error: malformed :: The request message was malformed :: Error creating new registration :: No MX record for domain yaxom.com

This error is plain wrong. There are 7 MX records for the yaxom.com domain.
Can this be fixed?

**Comment by kuba** [comment]
@gjb which email address did you use, exactly? @jcjones, can you have a look?

**Comment by kuba** [comment]
I'm actually experiencing the same problem for `kuba.us.to` and staging API endpoint. Likely to be CA problem. https://letsencrypt.status.io/ doesn't seem to mention anything related...

**Comment by gjb** [comment]
@kuba The email address I used was gjb@yaxom.com

**Comment by jsha** [comment]
The error messages from Boulder on DNS failures are currently bad. We've been working on fixing those this week.

Looking through logs, it looks like we got a connection refused from ns3.yaxom.net trying look up yaxom.com. Testing locally, I get connection timed out:

```
 $ dig mx yaxom.com @ns3.yaxom.net

; <<>> DiG 9.9.5-11ubuntu1-Ubuntu <<>> mx yaxom.com @ns3.yaxom.net
;; global options: +cmd
;; connection timed out; no servers could be reached
```

Note that ns1 and ns2 .yaxom.net appear to respond just fine.

**Comment by gjb** [comment]
@jsha @kuba Yes, ns3.yaxom.net is not working at present. But that's why there are three name servers, and the other two are responding correctly. Although it's true that I should fix the broken one, I might leave it for the time being as a test case. A single down name server should not prevent things from working.

**Comment by jsha** [comment]
Thanks for being willing to leave it up as a test case, much appreciated! We'll see what we can do to make it better.

**Comment by gjb** [comment]
@jsha Any progress on this? I don't want to leave that broken name server like that forever :)

**Comment by jsha** [comment]
@gjb: Sorry for the radio silence. We've resolved a number of DNS bugs in the last week. Can you try again now?

**Comment by gjb** [comment]
Thanks @jsha. There is progress, in that I was able to get letsencrypt-auto to complete creation of a cert. But it's still broken, as it would not accept my email address. I've attached a screenshot of the dialog with the complaint about my yaxom.com address. I worked around it for now by providing a gmail.com address, but that's not going to fly in production.

The error does not say what the problem was, so I don't know if it was still the MX record problem, but not as badly handled as before, or if it was something else -- "There seem to be problems with that address" is not very informative.

![screenshot from 2015-11-25 10 12 47](https://cloud.githubusercontent.com/assets/167270/11384810/c43b7480-935e-11e5-965d-774671f81d5f.png)
